runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldSwitcherPanel.java

```diff
@@ -230,23 +253,88 @@ void populate(List<World> worlds)
 		for (int i = 0; i < worlds.size(); i++)
 		{
 			World world = worlds.get(i);
+			EnumSet<WorldType> types = world.getTypes();
 
-			switch (filterMode)
+			switch (subscriptionFilterMode)
 			{
 				case FREE:
-					if (world.getTypes().contains(WorldType.MEMBERS))
+					if (types.contains(WorldType.MEMBERS))
 					{
 						continue;
 					}
 					break;
 				case MEMBERS:
-					if (!world.getTypes().contains(WorldType.MEMBERS))
+					if (!types.contains(WorldType.MEMBERS))
 					{
 						continue;
 					}
 					break;
 			}
 
+			if (exclusiveTypeFilter)
+			{
+				//this removes worlds that have no type as they will not match any filter(s)
+				EnumSet<WorldType> wt = EnumSet.allOf(WorldType.class);

```

Comment:
No need for the shorthand, give this variable a clearer name. Maybe `allWorldTypes`?